[Kelly Wang] iP

kellywsq03 opened the pull request:
## ECHO: Your one-stop planner 😱

> Echo changes your **to-dos** to _to-dones_ ;)

Become a [productivity pro](https://ca.indeed.com/career-advice/career-development/daily-to-do-list)!

Why use to-do GUIs that looks like this:
- [x] Walk dog
- [ ] Wash clothes
- [ ] Go for a run

When you can CHAT with ECHO!!
Chat with Echo to plan tasks as a:
- Todo
- Event
- Deadline

Just follow these steps:
1. Download the build
2. Run the build with gradle
3. Have fun!!

Take a sneak peak at our code, here's the `main` method:
```/**
     * The main method that launches the Echo application.
     *
     * @param args Command-line arguments (not used).
     */
    public static void main(String[] args) {
        new Echo("src/main/data/savedTasks.txt").run();
    }

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

kellywsq03 committed:
Modified chatbot's name to Echo. Modified welcome message.

kellywsq03 committed:
Chatbot echos user input. Chatbot exits when user inputs bye.

kellywsq03 committed:
User can add tasks. User can list tasks

kellywsq03 committed:
Added Tasks class. Mark and unmark behaviours added

kellywsq03 committed:
Added error handling. Differentiates Todos, Events, Deadlines.

kellywsq03 committed:
Tested todo, deadline and events functions

kellywsq03 committed:
Implemented deletion

kellywsq03 committed:
Used Enums

kellywsq03 committed:
Save tasks to file. Load tasks from file

kellywsq03 committed:
Merge branch 'branch-Level-7'

kellywsq03 committed:
Save deadline as DateTime

kellywsq03 committed:
Merge branch 'branch-Level-8'

kellywsq03 committed:
Make the code more OOP

kellywsq03 committed:
Merge branch 'branch-A-MoreOOP'

kellywsq03 committed:
Move classes into packages

kellywsq03 committed:
Merge branch 'branch-A-Packages'

kellywsq03 committed:
Merge branch 'add-gradle-support'

kellywsq03 committed:
Add Gradle

kellywsq03 committed:
Configure build.gradle

kellywsq03 committed:
Test TaskList and Parser

kellywsq03 committed:
Test TaskList and Parser

kellywsq03 committed:
Edit Hello message

kellywsq03 committed:
Merge branch 'branch-A-JUnit'

kellywsq03 committed:
Create JAR file

kellywsq03 committed:
Add JavaDoc headers to Ui

kellywsq03 committed:
Add JavaDoc header comments for Parser

kellywsq03 committed:
Add JavaDoc headers for Storage

kellywsq03 committed:
Add Javadoc headers to TaskList

kellywsq03 committed:
Add JavaDoc hbeaders to Echo, EchoException and TaskType

kellywsq03 committed:
Abide by coding conventions

kellywsq03 committed:
Add Find command

kellywsq03 committed:
Fix indentation

kellywsq03 committed:
Merge JavaDoc to CodingStandard

kellywsq03 committed:
Merge Level-9 to master

Brendan8899 reviewed (commented):
Good job defining your commands as Enumerations as commands belong to a fixed set. The capitalisation of command names is appropriate as constant names must all be uppercase

kellywsq03 committed:
Configure UI

kellywsq03 committed:
Improve GUI

kellywsq03 committed:
Merge branch 'branch-level-10'

LOLIPOP-INTELLIGENCE commented on src/main/java/echo/Echo.java:
```diff
@@ -0,0 +1,44 @@
+        return ui.handleInput(input);
+    }
+
+    public void stopRunning() {
```
Missing header comments

LOLIPOP-INTELLIGENCE commented on src/main/java/echo/Echo.java:
```diff
@@ -0,0 +1,44 @@
+        storage.saveToFile();
+    }
+
+    public String greetUser() {
```
Missing header comments.

LOLIPOP-INTELLIGENCE commented on src/main/java/echo/Ui.java:
```diff
@@ -0,0 +1,295 @@
+package echo;
```
The import statements should be reordered according to the coding standard. They should be grouped as follows: static imports (if any), Java imports, third-party imports (if any), then project imports.

LOLIPOP-INTELLIGENCE commented on src/main/java/echo/backend/Parser.java:
```diff
@@ -0,0 +1,187 @@
+            try {
+                LocalDate localDate = LocalDate.parse(s, formatter);
+                return localDate;
+            } catch (DateTimeParseException e) {
```
Consider adding a comment why the handling of this exception is ignored.

kellywsq03 committed:
Add assertions

Program assumes that all errors are handled and inputs are correct.

Add assertions to all relevant files.

Assertions are used to define assumptions about the program state so that the runtime can verify them.
This will highlight bugs in the code.

Refer to this document on Java assertions.
https://www.javatpoint.com/assertion-in-java

kellywsq03 committed:
Improve code quality

Code is written hastily. It may be difficult to read. Improve code quality by removing long methods, nested code and applying SLAP. This improves understanding and readibility

kellywsq03 committed:
Improve code quality

Code has nested methods and arrowhead-style code. Code is hard to read. Improve code readibility.

kellywsq03 committed:
Merge pull request #1 from kellywsq03/branch-A-Assertions

Add assertions

kellywsq03 committed:
Resolve merge conflicts

kellywsq03 committed:
Merge pull request #2 from kellywsq03/branch-A-CodeQuality

Branch a code quality

kellywsq03 committed:
Add update command

kellywsq03 committed:
Add update error handling

kellywsq03 committed:
Add User Guide

kellywsq03 committed:
Change image size in README

kellywsq03 committed:
Update User Guide

kellywsq03 committed:
Fix error handling. Update README

kellywsq03 committed:
Make long methods concise

kellywsq03 committed:
Fix code quality

kellywsq03 committed:
Remove very-long methods

kellywsq03 committed:
Fix filePath errors
